salt/_modules/caasp_hosts.py

```diff
@@ -155,15 +154,34 @@ def _add_names(hosts, ips, names):
             hosts[ip] = _concat(hosts[ip], names)
 
 
+def _prepend_names(hosts, ips, names):
+    if not isinstance(names, list):
+        names = [names]
+    if not isinstance(ips, list):
+        ips = [ips]
+
+    for ip in ips:
+        __utils__['caasp_log.debug']('hosts: adding %s -> %s', ip, names)
+        if ip not in hosts:
+            hosts[ip] = names
+        else:
+            hosts[ip] = names + hosts[ip]
```

Comment:
```Python
def _add_names(hosts, ips, names, fun):
    if not isinstance(names, list):
        names = [names]
    if not isinstance(ips, list):
        ips = [ips]
    for ip in ips:
        __utils__['caasp_log.debug']('hosts: adding %s -> %s', ip, names)
        if ip not in hosts:
            hosts[ip] = fun([], names)
        else:
            hosts[ip] = fun(hosts[ip], names)
```

so we could define 

```Python
def _prepend(lst1, lst2):
    res = list(set(lst2) | set(lst1))  # join both lists (without dups)
    res = [x for x in res if x]  # remove empty strings
    return res
```

(we should even calle these functions `prepend` and `append` for coherency)

use `_add_names(hosts, ip, [id, id + '.' + infra_domain], _concat)` or `_add_names(hosts, ip, [nodename, nodename + '.' + infra_domain], _prepend)`...

Reply:
`res = list(set(lst2) | set(lst1))  # join both lists (without dups)` is itself problematic, because the order preservation in `lst2` is not guaranteed. The fact that we are using `set` to remove duplicates also makes the order inside `lst2` subject to change.

For our specific usecase it's not an issue, because `nodename` and `nodename.infra.caasp.local` is the only usage of this, and order will always come with `nodename` first (the desired result), but this is based on language implementation of `set` and this can break again. OTOH it's weird to change `_prepend` to not use a `set` while `_concat` or the first argument of `_prepend` does.